agent/base

```diff
@@ -308,5 +308,5 @@ function interrupt() {
     if [[ ${?} != 0 ]]; then
         error_log "[${script_name}]: failed to stop the tool meisters on interrupt."
     fi
-    return 0
+    exit 0
```

Comment:
Is this fix worth factoring out to its own commit?

Reply:
It's all part of handling ^C relatively gracefully; none of these elements are an end-to-end fix alone, so it didn't seem worth the extra effort of factoring out pieces.